File: web-home/src/main/webapp/WEB-INF/generator/template/webgenerator/dao/spring_jdbc/deprecated/java/${basepackage_dir}/webservice/${className}WebService.java
<#include "/java_copyright.include">
<#assign className = table.className>   
<#assign classNameLower = className?uncap_first> 
package ${basepackage}.webservice;


import ${basepackage}.model.*;
import ${basepackage}.query.*;

import com.github.rapid.common.util.page.Page;
import com.github.rapid.common.util.page.PageQuery;
import java.util.Date;
import java.util.List;

/**
 * [${className}] 的WebService接口
 * 
 * @author ${author}
 * @version 1.0
 * @since 1.0
 */
public interface ${className}WebService {

	/** 
	 * 创建${className}
	 **/
	public void create(${className} ${classNameLower});
	
	/** 
	 * 更新${className}
	 **/	
    public void update(${className} ${classNameLower});
    
	/** 
	 * 删除${className}
	 **/
    public void remove(<@generateArguments table.pkColumns/>);
    
	/** 
	 * 根据ID得到${className}
	 **/    
    public ${className} getById(boolean join,<@generateArguments table.pkColumns/>);
    
	/** 
	 * 分页查询: ${className}
	 **/      
	public Page<${className}> query(boolean join,${className}Query query);
	
	/** 
	 * 搜索: ${className}
	 **/ 
	public List<${className}> search(boolean join,String query,PageQuery pageQuery);
}

File: web-home/src/main/webapp/WEB-INF/generator/template/webgenerator/dao/spring_jdbc/deprecated/ui/js/app/${namespace}/${classNameLowerCase}/${className}WebService.js
<#include "/macro.include"/>
<#include "/custom.include"/>  
<#assign className = table.className>   
<#assign classNameFirstLower = className?uncap_first> 
<#assign classNameLowerCase = className?lower_case>
"use strict";
/**
 * webservice for ${className}
 * author: ${author}
 */
var globalWsUrl = globalWsUrl ? globalWsUrl : '/rpc';
var ${classNameFirstLower}WsUrl = globalWsUrl + "/${className}WebService";
var ${className}WebService = {

	query : function(query,success,error) {
		var q = ${className}Util.convertQuery(query);
		$.jsonp(${classNameFirstLower}WsUrl+"/query",q,AppUtil.wsCallback(success,error));
	},
	
	getone : function(<@generatePassingParameters table.pkColumns/>,success,error) {
		var pk = <#compress>{<#list table.pkColumns as column>${column.columnNameLower} : ${column.columnNameLower}<#if column_has_next>,</#if></#list>};</#compress>
		$.jsonp(${classNameFirstLower}WsUrl+"/getone",pk,AppUtil.wsCallback(success,error));
	},
	
	remove : function(<@generatePassingParameters table.pkColumns/>,success,error) {
		var pk = <#compress>{<#list table.pkColumns as column>${column.columnNameLower} : ${column.columnNameLower}<#if column_has_next>,</#if></#list>};</#compress>
		$.jsonp(${classNameFirstLower}WsUrl+"/remove",pk,AppUtil.wsCallback(success,error));
	},
	
	create : function(${classNameFirstLower},success,error) {
		var entity = ${className}Util.convertEntity(${classNameFirstLower});
		$.jsonp(${classNameFirstLower}WsUrl+"/create", entity,AppUtil.wsCallback(success,error));
	},
	
	update : function(${classNameFirstLower},success,error) {
		var entity = ${className}Util.convertEntity(${classNameFirstLower});
		$.jsonp(${classNameFirstLower}WsUrl+"/update", entity,AppUtil.wsCallback(success,error));
	}
	
};